docs/csharp/tutorials/ranges-indexes.md

```diff
@@ -69,7 +69,15 @@ The following sample shows many of the reasons for those choices. Modify `x`, `y
 
 [!code-csharp[SemanticsExamples](~/samples/csharp/tutorials/RangesIndexes/IndicesAndRanges.cs#IndicesAndRanges_Semantics)]
 
-## Scenarios for Indices and Ranges
+## Type support for indices and ranges
+
+If a type provides an [indexer](../programming-guide/indexers/index.md) with an <xref:System.Index> or <xref:System.Range> parameter, it explicitly supports indices or ranges respectively.
+
+A type is **countable** if it has a property named `Length` or `Count` with an accessible getter and a return type of `int`. A countable type that doesn't explicitly support indices or ranges may provide an implicit support for them. For more information, see the [Implicit Index support](~/_csharplang/proposals/csharp-8.0/ranges.md#implicit-index-support) and [Implicit Range support](~/_csharplang/proposals/csharp-8.0/ranges.md#implicit-range-support) sections of the [feature proposal note](~/_csharplang/proposals/csharp-8.0/ranges.md).
```

Comment:
The links in this paragraph assume that  dotnet/csharplang#2797 is merged. Is that fine?

Reply:
Yes, that's fine.  I'll wait to merge this until the related PR is merged. I'm hoping that happens soon. There are some delays as teams are focused on .NET Conf.